server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1747,8 +1747,7 @@ public void close(String reason, boolean flushEngine) throws IOException {
                 } finally {
                     // playing safe here and close the engine even if the above succeeds - close can be called multiple times
                     // Also closing refreshListeners to prevent us from accumulating any more listeners
-                    // Closing remoteStore as a part of IndexShard close. null check is handled by IOUtils
-                    IOUtils.close(engine, globalCheckpointListeners, refreshListeners, pendingReplicationActions, remoteStore);
+                    IOUtils.close(engine, globalCheckpointListeners, refreshListeners, pendingReplicationActions);
```

Comment:
can you point to the code or add description on where the remoteStore close is handled?

Reply:
We are not closing `remoteStore` instance. I have added a description explaining why it is safe not to close `remoteStore`.

Reply:
Is it possible to not have the underlying lock being shared. I see we have created an issue for composite store. Do we know what is this lock used for?

Reply:
[ShardLock](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/env/ShardLock.java) is used as a lock. 

[NodeEnvironment](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/env/NodeEnvironment.java#L188) keeps map of shard ID to these locks.
We either modify locking logic or use CompositeStore.